cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -662,7 +662,7 @@ public void onBecomeOfflineFromFollower(Message message, NotificationContext con
         // "live_leader_or_follower")
         HelixAdmin admin = context.getManager().getClusterManagmentTool();
         ExternalView view = Utils.getHelixExternalViewWithFixedRetry(admin, cluster, resourceName);
-        Map<String, String> stateMap = view.getStateMap(partitionName);
+        Map<String, String> stateMap = Utils.getStateMap(view, partitionName);
```

Comment:
This happens in the FOLLOWER->OFFLINE state transition. 

What it does is to retrieve the upstream for the shard, and tell all other follower/offline replicas to use that upstream. If no upstream found in the stateMap, then no-op.

Therefore this PR change is straightforward, if the stateMap for the shard is empty, it means no upstream, so it's a no-op, and there is no need to make it a ERROR shard.